docs/docs/genworlds-framework/agents/agents.md

```diff
@@ -8,43 +8,72 @@ Agents are the entities that interact with the world. They have a set of goals a
-   
+
 3. **Remembering past events and relevant information:** Using its stored memories, the agent recalls past experiences and data that might affect its current decision-making process.
-   
+
 4. **Updating the plan and deciding actions:** Based on the world state, new events, and past memories, the agent updates its action plan and decides on the next actions. These could involve interacting with the world, other agents, or objects. Importantly, an agent can execute multiple actions in one step, improving overall efficiency.
-   
+
 5. **Executing the actions:** Finally, the agent implements its plan, influencing the world state and potentially triggering responses from other agents.
-   
+
 This interactive process fosters the emergence of complex, autonomous behavior, making each agent an active participant in the GenWorld.
 
 While we are currently focused on enhancing each of these steps, we foresee potential developments in the short-medium term. For instance, we're exploring the value and nature of "reflection" as an aspect of an agent's mental model. This would enable the agent to draw new conclusions from a set of recent memories and maintain high-level goals. We're also considering improvements to the communication systems between agents to facilitate more effective collaboration.
 
-## TreeAgent
+## BaseAgent
 
-The TreeAgent implements a version of Tree-of-Thought. It is the most powerful and reliable type of agent available in the Genworlds framework.
+The BaseAgent is the most powerful and reliable type of agent available in the Genworlds framework.
 
-It consists of a branching pipeline of multiple [Brains](./brains.md), that enable breaking down the task into smaller, more specific steps to improve the performance.
+It consists of a branching pipeline of multiple [Brains](/docs/genworlds-framework/agents/think.md), that enable breaking down the task into smaller, more specific steps to improve the performance.
 
 The first brain is always a navigation brain - it needs to select the next action for the agent.
 
 Based on the selected action type, a pipeline of brains is chosen. There can be any number of various brains in the pipeline.
 
 The output of each brain is passed to the next one, and the final brain needs to produce a complete set of parameters selected next action.
 
-### Custom memories 
+```mermaid
+graph TD
+    subgraph Simulation
+        W1(Simulation Socket)
+        subgraph World
+          subgraph Agent
+              P1(Listens)
+              P2(Interprets)
+              P3(Thinking Process)
+              P4(Action)
+              P5(Learns)
+              P1-->P2
+              P2-->P3
+              P3-->P4
+              P4-->P5
+          end
+            A1(Other Agents)
+            O1(Other Objects)
+        end
+    end
+  I1(Interfaces / APIs / Backends)
+  W1-->P1
+  P4-->W1
+  W1<-->A1
+  W1<-->O1
+  W1<-->I1
+  style Agent stroke:#f66,stroke-dasharray: 5 5, stroke-width:3px
+```
```

Comment:
Ok we can put it higher so is more understandable, but applies to all agents because they inherit from BaseAgent. For instance is not what the AutoGPT etc is doing.